src/components/swap/SwapModalFooter.tsx

```diff
@@ -141,3 +170,43 @@ export default function SwapModalFooter({
     </>
   )
 }
+
+function AnimatedLineItem(props: SwapLineItemProps & { open: boolean; delay: number }) {
+  const { open, delay } = props
+
+  const animatedProps = useSpring({
+    animatedOpacity: open ? 1 : 0,
+    config: { duration: ms('300ms'), easing: easings.easeOutSine },
+    delay,
+  })
+
+  return <SwapLineItem {...props} {...animatedProps} />
+}
+
+function AdvancedSwapDetails(props: { trade: InterfaceTrade; allowedSlippage: Percent; open: boolean }) {
```

Comment:
we have another `AdvancedSwapDetails` component defined in the project already - might get confusing to have both? can we rename this something like `ExpandableLineItems` ?

Reply:
excellent point as always edward